### ISSUE TITLE
Some issues on iPhone

### ISSUE THREAD

**Issue by xstortionist** [kticka/jquery-outclick#2]
Noticed that outclick not working as expected on IOS iphone and tablet.

http://celsiusmarketing.net/yuma/

If you open menu and click outside the menu it doesn't close. Works as expected on desktop and android tho.

**Comment by kticka** [comment]
I will check that asap, I'm android guy, so I dont have iphone at the moment :)

**Comment by xstortionist** [comment]
I'm an adroid guy as well. Just had everyone and their brother with an iPhone at the office tell me they couldn't close menu clicking outside. I tell them I'm not hip and rich like them and have way to test with iPhone! LOL.

**Comment by xstortionist** [comment]
For now, I just add click event to page overlay so they get off my back but I was using the outclick to close it.

**Comment by xstortionist** [comment]
Looks like mouseup doesn't work with ipad / iphones. Found on stackoverflow to use touchstart / touchend events instead.

**Comment by kticka** [comment]
Probably that would be good solution. But I don't want implement it until I can't test it on real device. I have tried few emulators, but it seems they doesn't repeat real ios behavior.

**Comment by xstortionist** [comment]
on Monday I'll have iOS device to test with and I'll play around with it. For now, I'm just going to add an additional click event to close menu.

**Comment by xstortionist** [comment]
I added this but no way to personally test it. This evening I'll be buying an unlocked iphone and will test on my end.

`enable: function() {
    return $(document).on('mouseup.outclick, touchend.outclick', this.trigger);
  },
  disable: function() {
    return $(document).off('mouseup.outclick, touchend.outclick');
  },`

**Comment by xstortionist** [comment]
FYI, just want to let you know what I just posted works. I had the people in the office test with and without touchend.outclick and that seems to fix the problem.

**Comment by xstortionist** [comment]
Got my hands on iphone 5 and tested sunday evening. Everything works as expected now. Not sure if you want to test on your end still but perhaps what I provided is not the best method. I just wanted to respond and let you know that it's working now. Thanks!

**Comment by kticka** [comment]
Hi there, sorry for delay. Well, touchend event will solve issue on iphone, but it brings other issue to android phones. When both actions are suitable, then outclick callback will be called twice. I need to solve this before commiting everything.

**Comment by kticka** [comment]
I will try to find best solution asap. I do not want to put any timers there (after event to avoid double fire), be cause it will enable timers everytime you click anywhere. That's main difference between click/outclick.

**Comment by xstortionist** [comment]
@kticka makes sense with the double firing issue that you speak of. For whatever reason, when using touchend.outclick only, it doesn't work on android but it does work on iphone. It doesn't seem that touchend works on android.

**Comment by xstortionist** [comment]
@kticka any luck being able to test this on your end?

**Comment by kticka** [comment]
Sorry, I was very busy last few weeks. I will add your solution in few days. Probably we can only have problem on devices that have both click and touch features (laptops with touchscreen, etc.). But I assume this isn't very common situation, so for now i'll put both, mouseup and touchend events.
